diff --git a/include/packio/internal/log.h b/include/packio/internal/log.h
index 1de6fb1..fafa71d 100644
--- a/include/packio/internal/log.h
+++ b/include/packio/internal/log.h
@@ -5,19 +5,14 @@
 #ifndef PACKIO_LOG_H
 #define PACKIO_LOG_H
 
-#if defined(PACKIO_LOGGING)
-#include <spdlog/spdlog.h>
-#define PACKIO_TRACE(...) SPDLOG_TRACE(__VA_ARGS__)
-#define PACKIO_DEBUG(...) SPDLOG_DEBUG(__VA_ARGS__)
-#define PACKIO_INFO(...) SPDLOG_INFO(__VA_ARGS__)
-#define PACKIO_WARN(...) SPDLOG_WARN(__VA_ARGS__)
-#define PACKIO_ERROR(...) SPDLOG_ERROR(__VA_ARGS__)
-#else
-#define PACKIO_TRACE(...) (void)0
-#define PACKIO_DEBUG(...) (void)0
-#define PACKIO_INFO(...) (void)0
-#define PACKIO_WARN(...) (void)0
-#define PACKIO_ERROR(...) (void)0
-#endif
+#include "utils/packio_logger.h"
+
+using namespace plc::core;
+
+#define PACKIO_TRACE(...) PackIoLogger::getLogger()->trace(__VA_ARGS__)
+#define PACKIO_DEBUG(...) PackIoLogger::getLogger()->debug(__VA_ARGS__)
+#define PACKIO_INFO(...) PackIoLogger::getLogger()->info(__VA_ARGS__)
+#define PACKIO_WARN(...) PackIoLogger::getLogger()->warn(__VA_ARGS__)
+#define PACKIO_ERROR(...) PackIoLogger::getLogger()->error(__VA_ARGS__)
 
 #endif // PACKIO_LOG_H
diff --git a/include/packio/json_rpc/rpc.h b/include/packio/json_rpc/rpc.h
index 80ac6c1..e9f81dd 100644
--- a/include/packio/json_rpc/rpc.h
+++ b/include/packio/json_rpc/rpc.h
@@ -14,7 +14,7 @@
 #include "../internal/log.h"
 #include "../internal/rpc.h"
 #include "converters.h"
-#include "hash.h"
+// #include "hash.h"
 
 namespace packio {
 namespace json_rpc {
